dtfs-central-api/src/v1/controllers/portal/deal/create-deal.controller.js

```diff
@@ -41,23 +46,36 @@ const createDeal = async (deal, maker) => {
 };
 
 exports.createDealPost = async (req, res) => {
-  const { user } = req.body;
-  if (!user) {
-    return res.status(400).send({ status: 400, message: 'Invalid user' });
-  }
+  try {
+    const { user, deal, auditDetails } = req.body;
 
-  if (typeof req?.body?.deal?.dealType !== 'string') {
-    return res.status(400).send({ status: 400, message: 'Invalid deal type' });
-  }
+    if (!user) {
+      return res.status(400).send({ status: 400, message: 'Invalid user' });
+    }
 
-  const { validationErrors, _id } = await createDeal(req.body.deal, user);
+    if (typeof deal?.dealType !== 'string') {
+      return res.status(400).send({ status: 400, message: 'Invalid deal type' });
+    }
 
-  if (validationErrors) {
-    return res.status(400).send({
-      _id,
-      validationErrors,
-    });
-  }
+    validateAuditDetailsAndUserType(auditDetails, 'portal');
+
+    const { validationErrors, _id } = await createDeal(deal, user, auditDetails);
 
-  return res.status(200).send({ _id });
+    if (validationErrors) {
+      return res.status(400).send({
+        _id,
+        validationErrors,
+      });
+    }
+
+    return res.status(200).send({ _id });
+  } catch (error) {
```

Comment:
fwiw I'd been wrapping the try catch around `validateAuditDetails` only, doesn't really matter at all though :)

Reply:
Yeah I've moved this for now